src/pfe/file-watcher/server/test/scripts/exec.sh

```diff
@@ -0,0 +1,114 @@
+WEBSERVER_FILE="$CW_DIR/src/pfe/iterative-dev/test/microclimate-test/scripts/webserver.sh"
+
+function usage {
+    me=$(basename $0)
+    cat <<EOF
+Usage: $me: [-<option letter> <options value> | -h]
+Options:
+    -t # Test type, currently support 'local' and 'kube' - Mandatory
+    -s # Test suite, currently support 'functional' - Mandatory
+    -h # Display the man page
+EOF
+}
+
+function clone () {
+    curl -H "Content-type: application/json" -d '{"projectName": "'$1'", "parentPath": "'$2'", "url": "'$3'"}' "http://localhost:9090/api/v1/projects"
+}
+
+function setup {
+    DATE_NOW=$(date +"%d-%m-%Y")
+    TIME_NOW=$(date +"%H.%M.%S")
+    BUCKET_NAME=turbine-$TEST_SUITE
+    TEST_DIR=$CW_DIR/src/pfe/file-watcher/server
+    TEST_DIR_CONTAINER=/file-watcher/server
+    TEST_OUTPUT_DIR=~/test_results/$TEST_SUITE/$DATE_NOW/$TIME_NOW
+    TEST_OUTPUT=$TEST_OUTPUT_DIR/test_output.xml
+    TEST_LOG=$TEST_OUTPUT_DIR/$TEST_SUITE-test.log
+    CODEWIND_CONTAINER_ID=$(docker ps | grep codewind-pfe-amd64 | cut -d " " -f 1)
+
+    mkdir -p $TEST_OUTPUT_DIR \
+    && docker exec -it $CODEWIND_CONTAINER_ID bash -c "cd /file-watcher/server; npm install" \
+    && docker exec -it $CODEWIND_CONTAINER_ID bash -c "cd /file-watcher/server; npm install --only=dev" \
+    && docker cp $TEST_DIR/test $CODEWIND_CONTAINER_ID:$TEST_DIR_CONTAINER
+
+    if [[ $? -ne 0 ]]; then
+        echo -e "${RED}Setup is failed. ${RESET}\n"
+        exit 1
+    fi
+    
+    # Clone projects to workspace
+    declare -A PROJECT_URLS
+    PROJECT_URLS["codewindtestpython"]="https://github.com/microclimate-dev2ops/SVTPythonTemplate"
+    PROJECT_URLS["codewindtestgo"]="https://github.com/microclimate-dev2ops/microclimateGoTemplate"
+    PROJECT_URLS["codewindtestlagom"]="https://github.com/microclimate-dev2ops/lagomJavaTemplate"
+    PROJECT_URLS["testspring"]="https://github.com/microclimate-dev2ops/springJavaTemplate"
+    PROJECT_URLS["testmicroprofile"]="https://github.com/microclimate-dev2ops/javaMicroProfileTemplate"
+    PROJECT_URLS["testnodejs"]="https://github.com/microclimate-dev2ops/nodeExpressTemplate.git"
+    PROJECT_URLS["testswift"]="https://github.com/microclimate-dev2ops/swiftTemplate"
+    for PROJECT_NAME in ${!PROJECT_URLS[@]}; do
+        clone $PROJECT_NAME $CW_DIR/codewind-workspace ${PROJECT_URLS[$PROJECT_NAME]}
+    done
+}
+
+function run {
+    docker exec -it $CODEWIND_CONTAINER_ID bash -c "cd /file-watcher/server; JUNIT_REPORT_PATH=/test_output.xml npm run $TEST_SUITE:test:xml" | tee $TEST_LOG
+    docker cp $CODEWIND_CONTAINER_ID:/test_output.xml $TEST_OUTPUT
+    echo -e "${BLUE}Test logs available at: $TEST_LOG ${RESET}"
+
+    if [[ -z $DEVELOPMENT_RUN ]]; then
+        echo -e "${BLUE}Upload test results to test dashboard. ${RESET}"
+        $WEBSERVER_FILE $TEST_OUTPUT_DIR > /dev/null
+        curl --header "Content-Type:text/xml" --data-binary @$TEST_OUTPUT --insecure "https://$DASHBOARD_IP/postxmlresult/$BUCKET_NAME/test" > /dev/null
```

Comment:
You should check if `DASHBOARD_IP` is present if `DEVELOPMENT_RUN` is empty to handle error case

Reply:
Actually the curl command will throw error by itself if no `DASHBOARD_IP`, but I can add code to handle the check explicitly